google-storage.yml

```diff
@@ -107,6 +107,30 @@ provision:
       For more information about autoclass feature, see https://cloud.google.com/storage/docs/autoclass
     default: false
     prohibit_update: true
+  - field_name: retention_policy_retention_period
+    type: number
+    details: |
+      The period of time, in seconds, that objects in the bucket must be retained and cannot be deleted, overwritten, or archived,
+      in other words, it is the configuration of the bucket's data retention policy for how long objects in the bucket should be retained.
+      The value must be less than 2_147_483_647 seconds
+      For more information about retention policy feature, see https://cloud.google.com/storage/docs/bucket-lock
+    default: 0
+    constraints:
+      minimum: 0
+  - field_name: retention_policy_is_locked
+    type: boolean
+    details: |
+      The `retention_policy_is_locked` property locks a retention policy to permanently set it on the bucket.
+      A retention policy locked implies:
+      * It is not possible to remove it.
+      * It is not possible to delete a bucket unless every object in the bucket has met the retention period.
+      * It is not possible to reduce the retention period of a locked retention policy.
+      * It is not possible to increase the retention period of a locked retention policy.
+      Caution: Lock a retention policy is an irreversible action. Once set to true, it can no longer be set to false.
+      This change will attempt to recreate the bucket, but the system will block it and send an error.
```

Comment:
I think this phrase could cause some concern and folks might ask for more information:
> This change will attempt to recreate the bucket, but the system will block it and send an error.

Perhaps something like this may be sufficient:
> Caution: Lock a retention policy is an irreversible action. Once set to `true`, any attempt to set it to `false` or to delete the service will result in an error.

Also "Locking" is better than "Lock" in the above sentence.

Reply:
@blgm, thank you very much for the suggestion!